### ISSUE TITLE
Performance regression: MvcJson2k

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/Benchmarks#1151]
A performance regression has been detected for the following scenarios:

| Scenario | Environment | Date | Old RPS | New RPS | Change | Deviation |
| -------- | ----------- | ---- | ------- | ------- | ------ | --------- |
| MvcJson2k | Linux, http, KestrelSockets | 2019-06-20 18:16:40Z | 234950 | 228819 | -2.61 % | -2.89 σ |

Before versions:
Microsoft.AspNetCore.App __3.0.0-preview7.19319.3__
Microsoft.NetCore.App __3.0.0-preview7-27819-07__

After versions:
Microsoft.AspNetCore.App __3.0.0-preview7.19319.3__
Microsoft.NetCore.App __3.0.0-preview7-27819-12__

Commits:

__Microsoft.NetCore.App / Core FX__
https://github.com/dotnet/corefx/compare/edc88b0fe55278ee43c821ab05707d8bb0befded...5ec88f81af71d253cfa74f9982abf748ccbd529b

__Microsoft.NetCore.App / Core CLR__
https://github.com/dotnet/coreclr/compare/eafa8648ebee92de1380278b15cd5c2b6ef11218...56d57afaab150be6d1cf39869a917efe9ec5ff43

**Comment by pranavkm** [comment]
FYI @steveharter \ @ahsonkhan  this is likely the converter change - https://github.com/dotnet/corefx/commit/6b41813015ad0db7740523d65acfcd6a6d7b2df5.

**Comment by steveharter** [comment]
The converter change did not introduce any perf hit in my local benchmarks with many iterations. Internally, we already used converters, and the only extra work we do now is during warm-up (reading attributes primarily).

What is the date range of 3.0.0-preview7-27819-07 to *-12? It may be that this caused a regression due to preventing "ordered" optimization for deserialization: https://github.com/dotnet/corefx/pull/38705. That code has since been replaced.
